test/unit/lib/input-selection.test.ts

```diff
@@ -0,0 +1,37 @@
+import { get as getCurrentSelection } from "../../../src/lib/input-selection";
+
+describe("getCurrentSelection", function () {
+  describe("element type", function () {
+    it("returns selection range for input", function () {
```

Comment:
Doing this makes the typescript compiler unhappy since it can't tell what type `document.createElement(elementType);`

Are you okay with just coercing them both to an `HTMLInputElement`?

```ts
import { get as getCurrentSelection } from "../../../src/lib/input-selection";

describe("getCurrentSelection", function () {
  describe("element type", function () {
    it.each(["input", "textarea"])(
      "returns selection range for %s",
      (elementType) => {
        const element = document.createElement(elementType) as HTMLInputElement;

        // set a value so the element has text to select
        element.value = "asdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasd";
        element.setSelectionRange(10, 23);

        const result = getCurrentSelection(element);

        expect(result).toEqual({
          start: 10,
          end: 23,
          delta: 13,
        });
      }
    );
  });
});
```

Reply:
for tests? sure. tbh that's arguably a better test. (also arguably not.)(i would not want to be in the room for that pedantic debate actually)

Reply:
Fair enough. Pushed up the change.